first-order-odes/numerical-solution.md

```diff
@@ -1 +1,98 @@
 # Numerical solution methods
+
+Some ODEs are hard (or impossible!) to solve analytically... or maybe you just
+don't want to! Numerical approximations to the solution can be very valuable in
+these cases.
+
+````{topic} Euler's method
+To solve the the first-order ODE $y' = f(t, y)$, approximate $y(t)$ using a
+Taylor series:
+
+```{math}
+:label: euler-method
+y(t + \Delta t) \approx y(t) + f(t, y) \Delta t
+```
+
+This update can be applied repeatedly to approximate *y* as a function of *t*.
+To start, *t* is initialized to $t_0$ and *y* is started from the initial
+condition $y_0 = y(t_0)$. Then, going from $n = 0$, advance $t_n$ and $y_n$
+
+\begin{align}
+t_{n+1} &= t_n + \Delta t \\
+y_{n+1} &= y_n + f(t_n, y_n)
+\end{align}
+
+until the desired value of *t* is reached.
+````
+
+For example, let's solve the initial value problem:
+
+```{math}
+y' - y = t, \qquad y(0) = 0
+```
+
+using Euler's method with a step size $\Delta t = 0.2$. First, rewrite in
+explicit form:
+
+```{math}
+ y' = t + y = f(t, y)
+```
+
+Then, form a table and update
+
+|  $n$ |  $t$   | $y$    | $f(t,y)$ |
+| ---- | ------ | ------ | -------- |
+| 0    | 0      | 0      | 0        |
+| 1    | 0.2    | 0      | 0.2      |
+| 2    | 0.4    | 0.04   | 0.44     |
+| 3    | 0.6    | 0.128  | 0.728    |
+| 4    | 0.8    | 0.274  | 1.074    |
+| 5    | 1.0    | 0.489  |          |
+
+Compare to Exact Solution
+
+```{math}
+F(x) = e^{\int -1 \, dx} = e^{-x}
+```
+
+```{math}
+\int x e^{-x} dx = -x e^{-x} + \int e^{-x} dx = -(x + 1) e^{-x}
+```
+
+```{math}
+y(x) = e^x \left[ - (x + 1) e^{-x} + C \right] = e^x - (x + 1)
+```
+
+```{math}
+0 = y(0) = -1 + C \Rightarrow C = 1
+```
+
+```{math}
+:label: exact-final
+y(x) = e^x - (x + 1)
+```
```

Comment:
Please update this section to include a bit more description of the steps. See `first-order-odes/integrating-factors.md` on the main branch:

https://github.com/mphowardlab/essential-math/blob/main/first-order-odes/integrating-factor.md

for an example to follow.